doc/modules/preprocessing.rst

```diff
@@ -240,10 +240,61 @@ Centering kernel matrices
 -------------------------
 
 If you have a kernel matrix of a kernel :math:`K` that computes a dot product
-in a feature space defined by function :math:`\phi`,
-a :class:`KernelCenterer` can transform the kernel matrix
-so that it contains inner products in the feature space
-defined by :math:`\phi` followed by removal of the mean in that space.
+in a feature space (possibly implicitly) defined by a function
+:math:`\phi(\cdot)`, a :class:`KernelCenterer` can transform the kernel matrix
+so that it contains inner products in the feature space defined by
+:math:`\phi(\cdot)` followed by removal of the mean in that space.
```

Comment:
Alternatively, this can be phrased easily if we introduce the notion of "centered Gram matrix" (see for instance, slide 71 of [the Kernel Course notes from J.Mairal and J.P.Vert](http://members.cbio.mines-paristech.fr/~jvert/svn/kernelcourse/slides/master2017/master2017.pdf)?
```
:class:`KernelCenterer` compute the centered Gram matrix associated to a
positive semidefinite kernel :math:`K`.
```

Reply:
I would be in favour of incorporating both formulation